src/app/tests/MockAclTestContext.h

```diff
@@ -0,0 +1,40 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <messaging/tests/MessagingContext.h>
+
+namespace chip {
+namespace Test {
+
+/**
+ * @brief The context of test cases for messaging layer. It wil initialize network layer and system layer, and create
+ *        two secure sessions, connected with each other. Exchanges can be created for each secure session.
+ */
+class MockAclTestContext : public LoopbackMessagingContext
```

Comment:
Why do you need to extend this class? Why not just over-ride the access control impl in your test itself?

Reply:
Moreover, creating this separate bespoke test context implies re-usability by other tests, when in fact, it's just meant to be used by your TestAclEvent.cpp test. So I'd rather it not be in a separate file by itself.

Reply:
if you see https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/AppTestContext.cpp#L42, that init has super::init and SetAccessControl, I cannot override the access control delegate here. I have to create MockAclTestContext

Reply:
maybe I can create the new API in AccessControl so that I can setDelegate in test.

Reply:
per offline chat with @bzbarsky-apple, his suggestion is
1. to have API to make delegate configurable?
2. Alternately, since this is the only consumer of GetPermissiveAccessControlDelegate maybe adding the API to that as my previous commit did is OK and we should just rename it to TestAccessControlDelegate

Reply:
then I would go back to method 2, and rename it.